docs/source/scheduling-state.rst

```diff
@@ -60,33 +60,41 @@ Tasks flow along the following states with the following allowed transitions:
 .. image:: images/task-state.svg
     :alt: Dask scheduler task states
 
-*  *Released*: Known but not actively computing or in memory
-*  *Waiting*: On track to be computed, waiting on dependencies to arrive in
-   memory
-*  *No-worker*: Ready to be computed, but no appropriate worker exists
-   (for example because of resource restrictions, or because no worker is
-   connected at all).
-*  *Processing*: All dependencies are available and the task is assigned to a
-   worker for compute (the scheduler doesn't know whether it's in a worker
-   queue or actively being computed).
-*  *Memory*: In memory on one or more workers
-*  *Erred*: Task computation, or one of its dependencies, has encountered an error
-*  *Forgotten* (not actually a state): Task is no longer needed by any client
-   or dependent task
+released
+    Known but not actively computing or in memory
+waiting
+    On track to be computed, waiting on dependencies to arrive in memory
+no-worker
+    Ready to be computed, but no appropriate worker exists (for example because of
+    resource restrictions, or because no worker is connected at all).
+processing
+    All dependencies are available and the task is assigned to a worker for compute (the
+    scheduler doesn't know whether it's in a worker queue or actively being computed).
+memory
+    In memory on one or more workers
+erred
+    Task computation, or one of its dependencies, has encountered an error
+forgotten
+    Task is no longer needed by any client or dependent task, so it disappears from the
+    scheduler as well. As soon as a task reaches this state, it is immediately
+    dereferenced from the scheduler.
+
+.. note::
+    There's no intermediate state between ``waiting`` / ``no-worker`` and `processing``:
+    as soon as a task has all of its dependencies in memory somewhere on the cluster, it
+    is immediately assigned to a worker. This can lead to very long task queues on the
+    workers, which are then rebalanced dynamically through :doc:`work-stealing`.
```

Comment:
This will no longer be valid after https://github.com/dask/distributed/issues/6631
CC @gjoseph92